File: vue3_apponintments/src/api/appointment.js
import request from "@/utils/request.js";

//值班展示列表
export const getdoctorschedule = (data) => {
  return request.post("/appointment/getdoctorschedule", data);
};

//获取科室信息
export const getDepartmentList = () => {
  return request.post("/admin/findAll");
};

//获取日期
export const getAppointmentDate = () => {
    return request.get("/appointment/getdate");
};

//点击预约进行预约的创建
export const getAppointmentAdd = (data) => {
    return request.post("/appointment/add", data);
};

//获取个人预约列表
export const getAppointment = (data) => {
  return request.post("/appointment/getappointment", data);
};

//获取个人预约列表
export const deleteAppointment = (data) => {
  return request.post("/appointment/delete", data);
};

File: gin_vue(5)/gin_vue/ruoters/appointment.go
package ruoters

import (
	"gin_vue/controllers"
	"github.com/gin-gonic/gin"
)

func AppointmentRuotersInit(r *gin.Engine) {
	patientGroup := r.Group("/appointment")
	{
		appointmentControllers := controllers.AppointmentControllers{}
		patientGroup.GET("/getdate", appointmentControllers.GetDate)                      //获取后七天日期
		patientGroup.POST("/add", appointmentControllers.AddAppointment)                  //进行预约（预约表的增加）
		patientGroup.POST("/delete", appointmentControllers.DeleteAppointment)            //取消预约（预约表的删除）
		patientGroup.POST("/getappointment", appointmentControllers.GetAppointment)       //获取预约表
		patientGroup.POST("/getdoctorschedule", appointmentControllers.GetDoctorsChedule) //获取医生值班表
		patientGroup.POST("/setappointment", appointmentControllers.SetAppointment)       //修改预约表状态（取消预约或完成或签到）
	}
}
